tensorflow/contrib/lite/examples/ios/simple/RunModelViewController.mm

```diff
@@ -32,188 +26,295 @@
-  self.urlContentTextView.text = inference_result;
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.imageView.image = [UIImage imageNamed:@"grace_hopper.jpg"];
 }
 
-@end
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
 
-// Returns the top N confidence values over threshold in the provided vector,
-// sorted by confidence in descending order.
-static void GetTopN(
-    const float* prediction,
-    const int prediction_size,
-    const int num_results, const float threshold,
-    std::vector<std::pair<float, int> >* top_results) {
-  // Will contain top N results in ascending order.
-  std::priority_queue<std::pair<float, int>,
-      std::vector<std::pair<float, int> >,
-      std::greater<std::pair<float, int> > > top_result_pq;
-
-  const long count = prediction_size;
-  for (int i = 0; i < count; ++i) {
-    const float value = prediction[i];
-
-    // Only add it if it beats the threshold and has a chance at being in
-    // the top N.
-    if (value < threshold) {
-      continue;
-    }
-
-    top_result_pq.push(std::pair<float, int>(value, i));
-
-    // If at capacity, kick the smallest value out.
-    if (top_result_pq.size() > num_results) {
-      top_result_pq.pop();
-    }
-  }
-
-  // Copy to output vector and reverse into descending order.
-  while (!top_result_pq.empty()) {
-    top_results->push_back(top_result_pq.top());
-    top_result_pq.pop();
-  }
-  std::reverse(top_results->begin(), top_results->end());
+- (IBAction)runModel:(UIButton *)sender {
+    self.textView.text = [self runInferenceOnImage:self.imageView.image];
 }
 
-NSString* FilePathForResourceName(NSString* name, NSString* extension) {
-  NSString* file_path = [[NSBundle mainBundle] pathForResource:name ofType:extension];
-  if (file_path == NULL) {
-    LOG(FATAL) << "Couldn't find '" << [name UTF8String] << "."
-	       << [extension UTF8String] << "' in bundle.";
-  }
-  return file_path;
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info; {
+    [picker dismissViewControllerAnimated:YES completion:nil];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        self.imageView.image = [self fixrotation:info[UIImagePickerControllerOriginalImage]];;
+    });
 }
 
-NSString* RunInferenceOnImage() {
-  std::string graph;
-  const int num_threads = 1;
-  std::string input_layer_type = "float";
-  std::vector<int> sizes = {1, 224, 224, 3};
+- (IBAction)openPhoto:(UIBarButtonItem *)sender {
+    UIImagePickerController *picker;
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypePhotoLibrary]) {
+        picker = [[UIImagePickerController alloc] init];
+        picker.delegate = self;
+        picker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    }
+    [self presentViewController:picker animated:YES completion:nil];
```

Comment:
should this line be in the block above? Not much point in presentViewController:nil?